versionsupport_v1_16_R3/src/main/java/com/andrei1058/bedwars/support/version/v1_16_R3/v1_16_R3.java

```diff
@@ -678,4 +686,83 @@ public void playRedStoneDot(Player player) {
     public void clearArrowsFromPlayerBody(Player player) {
         ((CraftLivingEntity)player).getHandle().getDataWatcher().set(new DataWatcherObject<>(11, DataWatcherRegistry.b),-1);
     }
+
+    @Override
+    public List<Block> calculateExplosionBlocks(IArena arena, Entity source, Location explosionLocation, float radius, boolean fire, BiFunction<Location, Block, Boolean> callback) {
+        HashSet<Block> blocks = new HashSet<>();
+        org.bukkit.World bukkitWorld = explosionLocation.getWorld();
+
+        net.minecraft.server.v1_16_R3.Entity sourceEntity = null;
+        if (source != null) {
+            sourceEntity = ((CraftEntity) source).getHandle();
+
+        }
+
+        net.minecraft.server.v1_16_R3.World world = ((CraftWorld) Objects.requireNonNull(bukkitWorld)).getHandle();
+
+        double locX = explosionLocation.getX();
+        double locY = explosionLocation.getY();
+
+        if (sourceEntity instanceof EntityTNTPrimed) {
+            locY = sourceEntity.e(0.0625D);
+        }
+
+        double locZ = explosionLocation.getZ();
+
+        Explosion explosion = new Explosion(world, sourceEntity, null, null, locX, locY, locZ, radius, fire, Explosion.Effect.BREAK);
+        ExplosionDamageCalculator calculator = (sourceEntity == null ? STATIC_DMG_CALC : new ExplosionDamageCalculatorEntity(sourceEntity));
+
```

Comment:
// Copied from Explosion#a() (The NMS implementation of https://minecraft.fandom.com/wiki/Explosion#Model_of_block_destruction)